src/openrct2-ui/windows/Park.cpp

```diff
@@ -1446,31 +1446,43 @@ static void window_park_objective_invalidate(rct_window* w)
     window_park_anchor_border_widgets(w);
 }
 
-/**
- *
- *  rct2: 0x0066945C
- */
-static void window_park_objective_paint(rct_window* w, rct_drawpixelinfo* dpi)
+static void window_park_objective_bars_paint(
+    float percentage, ScreenCoordsXY* screenCoords, rct_window* w, rct_drawpixelinfo* dpi, int32_t colour)
 {
-    WindowDrawWidgets(w, dpi);
-    window_park_draw_tab_images(dpi, w);
+    if (percentage > 1)
+    {
+        percentage = 1;
+    }
+    int16_t barLength = static_cast<int16_t>((221 - 4) * percentage);
+    ScreenRect* rect = new ScreenRect(screenCoords->x + 1, screenCoords->y + 1, screenCoords->x + 221 - 3, screenCoords->y + 9);
```

Comment:
Don't use pointers, prefer stack allocations:
```suggestion
    auto rect = ScreenRect(screenCoords->x + 1, screenCoords->y + 1, screenCoords->x + 221 - 3, screenCoords->y + 9);
```